transformations/gender_culture_diverse_name/transformation.py

```diff
@@ -0,0 +1,147 @@
+        for x in ents:
+            name = x.split()[0]
+            capito = name[0].isupper() # pun intended, hint: Italian
+            name = name.capitalize()
+            if name in self.name_all:
+                gender = self.name2gender[name]
+                country = self.name2country[name]
+                if self.retain_gender:
+                    gender_choose_from = gender
+                else:
+                    gender_choose_from = self.genders
+                if self.retain_culture:
+                    country_choose_from = country
+                else:
+                    country_choose_from = self.countries
+            
+                new_countries = random.choices(country_choose_from, k=n)
+                new_genders = random.choices(gender_choose_from, k=n)
+                new_names = [random.choice(self.names[c][n]) for c,n in zip(new_countries, new_genders)]
+                if not capito:
+                    new_names = [n.lower() for n in new_names]
+                
+                for new_name in new_names:
+                    ret.append(re.sub(r'\b%s\b' % re.escape(name), new_name, doc.text))
+                    ret_m.append((name, new_name))
+        
+        if len(ret) > max_output:
+            idxs = random.choices(range(len(ret)), k=max_output)
+            return [ret[idx] for idx in idxs], [ret_m[idx] for idx in idxs]
+        else:
+            return ret, ret_m
+                
+
+
+class gender_culture_diverse_name(SentenceOperation):
+    tasks = [TaskType.TEXT_CLASSIFICATION, TaskType.TEXT_TO_TEXT_GENERATION]
+    languages = ["en"]
+
+    def __init__(self, n=1, seed=0, max_output=1, retain_gender=False, retain_culture=False, data_path=None):
+        super().__init__(seed)
+        self.nlp = spacy.load("en_core_web_sm")
+        self.n = n
+        self.max_output = max_output
+
+        if data_path is None:
+            self.changer = change_gender_culture_diverse_name(
+                os.path.join(os.path.dirname(os.path.abspath(__file__)), 'data.json'),
+                retain_gender, 
+                retain_culture
+            )
+        else:
+            self.changer = change_gender_culture_diverse_name(
+                data_path,
+                retain_gender, 
+                retain_culture
+            )
+
+    def generate(self, sentence: str):
+        random.seed(self.seed + hash(sentence))
+        perturbed_texts, _ = self.changer.apply(self.nlp(sentence), self.n, self.max_output)
+
+        return perturbed_texts
+
+
+test = gender_culture_diverse_name()
+# sentence = 'Rachel Green, a sheltered but friendly woman, flees her wedding day and wealthy yet unfulfilling life.'
+# sentence = 'Phoebe Buffay is an eccentric masseuse and musician.'
+# sentence = 'Joey has many short-term girlfriends.'
+# sentence = 'Chandler Bing is a sarcastic and self-deprecating IT manager.'
+sentence = 'Monica was overweight as a child.'
+p = test.generate(sentence)
+print(p[0])
```

Comment:
Don't forget to remove these lines or put them in the ifmain, otherwise when importing the transformation they will be executed.

Reply:
Also applies to the two-way version

Reply:
Hi, thanks for the comment! fixed now.